src/collective/ifttt/tests/robot/test_iftttaction.robot

```diff
@@ -0,0 +1,123 @@
+# ============================================================================
+# IFTTTACTION ROBOT TESTS
+# ============================================================================
+#
+# Run this robot test stand-alone:
+#
+#  $ bin/test -s collective.ifttt -t test_iftttaction.robot --all
+#
+# Run this robot test with robot server (which is faster):
+#
+# 1) Start robot server:
+#
+# $ bin/robot-server --reload-path src collective.ifttt.testing.COLLECTIVE_IFTTT_ACCEPTANCE_TESTING
+#
+# 2) Run robot tests:
+#
+# $ bin/robot src/collective/ifttt/tests/robot/test_iftttaction.robot
+#
+#   for debug mode
+# $ bin/robot-debug src/collective/ifttt/tests/robot/test_iftttaction.robot
+
+
+#
+# See the http://docs.plone.org for further details (search for robot
+# framework).
+#
+# ============================================================================
+
+*** Settings ***
+
+Resource  plone/app/robotframework/selenium.robot
+Resource  plone/app/robotframework/keywords.robot
+Resource  Selenium2Screenshots/keywords.robot
+
+Library  Remote  ${PLONE_URL}/RobotRemote
+Library  Selenium2Screenshots
+
+Test Setup  Test Setup
+Test Teardown  Close all browsers
+
+
+
+*** Variables ***
+
+${SELENIUM2LIBRARY_RUN_ON_FAILURE}  Capture page screenshot
+${SCREENSHOTS}  false
+
+
+*** Test Cases ***
+
+Scenario: As a site administrator I can configure Ifttt Trigger Action on modified news item
+  Given I'm logged in as a Site Administrator
+   And I'm at Site Setup
+   And I'm at Content Rules
+   And I click add content rule
+   And I fill new rule form
+   And I select 'Object modified' into 'form-widgets-event' selectbox
+   And I press 'Save' clickbutton
+  When I can see Edit content rule
+   And I select 'plone.conditions.PortalType' into 'contentrules-add-condition' selectbox
+   And I press 'form.button.AddCondition' clickbutton
+   And I check 'Add Content Type Condition' on pagecontent
+   And I select 'News Item' into 'form-widgets-check_types' selectbox
+   And I press 'form-buttons-save' clickbutton
```

Comment:
@datakurre I've been stuck with this error. Kindly help
`Element is not clickable at point (x, y) Another element would receive the click`

Reply:
@Shriyanshagro The issues is that because of the modal, the page contains multiple similarly named buttons. The error comes from Selenium trying to click the button in the main page, which is currently covered by the overlay.

This worked for me:

    Click button  css=.pattern-modal-buttons #form-buttons-save

Remember that with modals (popups) you will sooner or later have timing issues (error because the popup has not yet been properly loaded). You can try to coup with those with keywords

    Page should contain  some text
    Page should contain element  css=.my-selector
    Wait until page contains  some text
    Wait until page contains element  css=.my-selector